data/the emerald sword.txt

```diff
@@ -0,0 +1,796 @@
+				goto next
+			
+			label fire
+			`	Distrustful, you immediately close the hatch and get back into your ship while they scatter.`
+				goto shoot
+			
+			label next
+			`	You get into the camp, shooting any mercenaries that remained hidden, and pillaging their tents. You find hundreds of thousands of credits worth of valuables, and eventually, you find the cube as well. With that in hand, you go back to your ship, ready to take it back to John.`
+				decline
+			
+			label spare
+			`	"Deal. Bring me the cube," you say.`
+			`	While one of them goes fetch it, you wait in an uncomfortable silence, watching their every move, while they watch yours. Eventually the young man comes back and hands the cube to you. You examine it, confirming it is the artifact you were looking for.`
+			`	"You have it, now go away. We won't get in your way again," she says, and the other mercenaries relax, going back to their tents.`
+			`	With the cube in hand, you have no more reason to stay here. You go back to your ship, ready to take it back to John.`
+				accept
+	
+	on decline
+		payment 853621
+
+mission "The Box 6b"
+	landing
+	name "The box"
+	description "Bring the cube back to John on Vinci."
+	source "Watcher"
+	destination "Vinci"
+	
+	to offer
+		has "The Box 5: done"
+	
+	on complete
+		conversation
+			`Carrying the mysterious box again, you make your way to John's lab. He lets you in, and you tell him the whole ballad of the lost cube.`
+			`	"So, retribution, huh? You think it was the officials from the Deep that got angry at us last time? Or maybe Foster's competitors for that dig site? I should have seen that coming."`
+			`	"Possibly" you say. I can't imagine any other reason."`
+			`	"At least you got it back, but I don't know if it would be wise to continue to work on it here, and I can't just leave the planet for a couple months with no consequences. I don't even have a clue on how to read that data yet."`
+			`	"And what, do you happen to have another shady contact who can help?" you ask.`
+			`	"Unfortunately not. And now we seem to be running against time. If only the aliens would be willing to work with us and give us the information we need to crack this box..."`
+			`	As soon as you hear that phrase, it suddenly hits you. You've met the Hai up north, and you heard from the Unfettered that they've met the Sheragi many millenia ago. Perhaps the Hai could be able to help.`
+			`	"I think I know what to do. You're not the only person with contacts in strange places."`
+			`	You pick up your things, placing the cube back inside the backpack.`
+			`	"Well, good luck with that, captain. Don't get in too much trouble."`
+		
+mission "The box 7a"
+	landing
+	name "The box"
+	description "Travel to Mirrorlake to visit the Hai museum of galactic history."
+	source
+		near "Ya Hai" 1 100
```

Comment:
Maybe you could make two separate missions with the same content but one of them checks if you have met the Hai and if so sends you directly there whereas the other is the one used here to send the player exploring if they haven't met the Hai. This way you still send the player who doesn't know the Hai exploring but the player who already does know them can just go straight there. It's a small change, but I feel it would be beneficial to the overall experience.

Reply:
Part of the mission relies on not only meeting the Hai, but also having done first contact with the Unfettered. That's the requirement for the questline to start. Since by that time in the mission chain it's implied that there are time constraints, i think it wouldn't fit it to just go exploring, it would require some major changes.